public/js/test/mochitest/browser_dbg_interrupts.js

```diff
@@ -0,0 +1,89 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/**
+ * Test resuming from button and keyboard shortcuts.
+ */
+
+const TAB_URL = EXAMPLE_URL + "doc-script-switching-01.html";
+const sourceURLs = [
+  "code-script-switching-01.js",
+  "code-script-switching-02.js"
+];
+const resumeKey = "VK_F8";
+
+add_task(function*() {
+  const dbg = yield initDebugger(TAB_URL, ...sourceURLs);
+  const { selectors: { getBreakpoints, getBreakpoint }, getState } = dbg;
+  const sources = [];
+  for (let url of sourceURLs) {
+    sources.push(findSource(dbg, url));
+  }
+
+  yield ensureAddBreakpoint(dbg, getState, getBreakpoint, sources[0].id, 5);
+  yield ensureAddBreakpoint(dbg, getState, getBreakpoint, sources[1].id, 6);
+  yield ensureAddBreakpoint(dbg, getState, getBreakpoint, sources[1].id, 7);
+  is(getBreakpoints(getState()).size, 3, "Three breakpoint exists");
+
+  yield testResumeButton(dbg);
+  yield testResumeKey(dbg);
+  yield testBreakOnNextState(dbg);
+});
+
+const ensureAddBreakpoint = Task.async(function* (dbg, getState, getBreakpoint,
+                                                  sourceId, line) {
+  yield addBreakpoint(dbg, sourceId, line);
+  ok(getBreakpoint(getState(), { sourceId, line }),
+    "Breakpoint has correct line");
+});
+
+const testResumeButton = Task.async(function* (dbg) {
+  info("Testing Resume Button");
+  ok(findElement(dbg, "button", "pause", "active"), "Pause button active");
+  invokeInTab("firstCall");
+  yield waitForPaused(dbg);
+  ok(findElement(dbg, "button", "resume", "active"), "Resume button enabled");
+  clickElement(dbg, "button", "resume", "active");
+  yield waitForThreadEvents(dbg, "resumed");
+  yield waitForPaused(dbg);
+
+  // Evaluate script while debugger is paused
+  const result = yield evaluate(dbg, "1+1")
+    .then(response => response.result);
+  is(result, 2, "Evaluated script while debugger paused");
+  yield ensureThreadResumed(dbg);
+});
+
+const testResumeKey = Task.async(function* (dbg) {
+  info("Testing Resume Key");
+  ok(findElement(dbg, "button", "pause", "active"), "Pause button disabled");
+  invokeInTab("firstCall");
+  yield waitForPaused(dbg);
+  ok(findElement(dbg, "button", "resume", "active"), "Resume button enabled");
+  pressKey(dbg, resumeKey);
+  yield waitForThreadEvents(dbg, "resumed");
+  yield waitForPaused(dbg);
+
+  // Evaluate script while debugger is paused
+  const result = yield evaluate(dbg, "1+1")
```

Comment:
I'm not sure I see the purpose of evaluating in this test. This doesn't really test anything related to keyboard shortcuts for stepping/resuming or the stepping buttons. This looks like it was ported over from the old test, but I think we can avoid the evaluate and adding the `evaluate` command for now.

Just trying to make the tests a little more focused than they were before.